evm/vm/message.py

```diff
@@ -187,12 +196,34 @@ def __init__(self,
             validate_access_list(access_list)
         self.access_list = access_list
 
+    @property
+    def paygas_total_fee(self):
+        return self.paygas_gasprice * self.transaction_gas_limit
+
+    @property
+    def paygas_gasprice(self):
+        if self._paygas_gasprice is None:
+            return 0
+        else:
+            return self._paygas_gasprice
+
+    @paygas_gasprice.setter
+    def paygas_gasprice(self, gas_price):
```

Comment:
Continuation of previous suggestion to remove the `setter` and just make this a method `set_PAYGAS_gasprice`.  Seems a bit more intuitive/correct/something.